### ISSUE TITLE
Endpoint for Navigation

### ISSUE THREAD

**Issue by cswertz** [cswertz/lerndorf#196]
The main navigation should show a list of learning units that can be accessed by the user with the current role. A list of knowledge units per learning unit can be opened within the menu. An endpoint that delivers the lists needed for the main navigation is needed. 
If the role is guest, all learning units that are visible for guests should be shown. If the role is user, all learning units created by that user and all learning units visible for logged in users should be shown. Further lists  for courses will be needed in the future.

**Comment by leganz** [comment]
## Given Briefing
```
a) if not CourseUser.sequenceId=UserId(user): wie Wissensbasis, zusätzlich:
1. KnowledgeType:
 - and KnowledgeUnit.Id=CourseContent.knowledgetUnit.id
 - and CourseContent.CourseId=CourseId (CourseId wurde durch Auswahl des Kurses gesetzt)
2. Media Type:
 - and KnowledgeUnit.Id=CourseContent.knowledgetUnit.id
 - and CourseContent.CourseId=CourseId (CourseId wurde durch Auswahl des Kurses gesetzt)

b if CourseUser.sequenceId=UserId(user): wie Wissensbasis, zusätzlich:
1. KnowledgeUnit.ID = CourseSequenceKnowledgeUnit.KnowledgeUnitId
2. and CourseUser.sequenceID=CourseSequenceKnowledgeUnit.Id
3. order by CourseSequenceKnowledgeUnit.prevId desc (Sequenzen bisher nicht vorhanden; Anlegen der Sequenzen siehe 4.1.3).

In diesem Fall über der Navigation einen Vorwärts- und Rückwärtsbutton einblenden, mit denen die Sequenz durchlaufen werden kann. Bei erster Seite Rückwärts-Button ausgrauen, bei letzter Seite Vorwärts-Button ausgrauen. Zwischen den Buttons Bezeichnung der Sequenz anzeigen. Wenn mehrere Sequenzen vorhanden sind mit Dropdown zum umschalten. Wenn umgeschaltet wird und die aktuell angezeigte Seite nicht in der neuen Sequenz
```

Regarding the the last paragraph: Is this still part of the ui concept? I can't find controls within the application. Not sure if that is part of the requirement

**Comment by leganz** [comment]
Should be resolved with #212

**Comment by cswertz** [comment]
It is.
